lib/acts_as_taggable_on_mongoid/taggable/changeable.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module ActsAsTaggableOnMongoid
+  module Taggable
+    # Overides of methods from Mongoid::Changeable
+    module Changeable
+      def tag_list_on_changed(tag_definition)
+        attribute_will_change!(tag_definition.tag_list_name)
+      end
+
+      def reload(*args)
+        tag_types.each_value do |tag_definition|
+          instance_variable_set tag_definition.all_tag_list_variable_name, nil
+          instance_variable_set tag_definition.tag_list_variable_name, nil
+        end
+
+        super(*args)
+      end
+
+      def changed
+        changed_values = super
+        tag_list_names = tag_types.values.map(&:tag_list_name).map(&:to_s)
+
+        changed_attributes.each_key do |key|
+          next unless tag_list_names.include?(key.to_s)
+
+          if public_send("#{key}_changed?")
+            changed_values << key unless changed_values.include?(key)
+          else
+            changed_values.delete(key)
+          end
+        end
+
+        changed_values
+      end
+
+      def changes
+        changed_values = super
+
+        tag_types.each_value do |tag_definition|
+          tag_list_name = tag_definition.tag_list_name
+
+          next unless changed_attributes.key? tag_list_name
+
+          changed_values[tag_list_name] = [changed_attributes[tag_list_name], public_send(tag_list_name)]
+        end
+
+        changed_values
+      end
+
+      def setters
```

Comment:
Remove tag_lists from the setters to prevent the fields from being injected into the model.